### PR TITLE
ERMS-2536, ERMS-2563

### PR DESCRIPTION
**Pull request by Moetez16**

### CODE REVIEW

File: app/grails-app/i18n/messages.properties [hunk 1]
```diff
@@ -1915,7 +1915,8 @@ subscription.details.basic.note.end = view to see them
 subscription.details.no_ents = No entitlements yet
 subscription.details.from_pkg = From Package
 subscription.details.from_pkg.all = All
-subscription.details.asAt = Entitlements as at
+subscription.details.asAt = Title on the key date
```
Comment (croqueGrec09) [inline app/grails-app/i18n/messages.properties:1918]: due date

File: app/grails-app/services/de/laser/SubscriptionService.groovy [hunk 2]
```diff
@@ -1707,9 +1707,9 @@ class SubscriptionService {
                     ((colMap.printIdentifierCol >= 0 && cols[colMap.printIdentifierCol].trim().isEmpty()) || colMap.printIdentifierCol < 0)) {
             } else {
 
-                TitleInstance tiObj = TitleInstance.executeQuery('select ti from TitleInstance ti join ti.ids ident where ident.ns in :namespaces and ident.value = :value', [namespaces:idCandidate.namespaces, value:idCandidate.value])
-                if (tiObj) {
-                    IssueEntitlement issueEntitlement = issueEntitlements.find { it.tipp.title.id == tiObj.id }
+                List<TitleInstance> tiObj = TitleInstance.executeQuery('select ti from TitleInstance ti join ti.ids ident where ident.ns in :namespaces and ident.value = :value', [namespaces:idCandidate.namespaces, value:idCandidate.value])
+                if (tiObj.size() > 0) {
+                    IssueEntitlement issueEntitlement = issueEntitlements.find { it.tipp.title.id in tiObj.id }
```
Comment (croqueGrec09) [inline app/grails-app/services/de/laser/SubscriptionService.groovy:1712]: Cannot work, tiObj is now an array and cannot have ids

Comment (Moetez16) [reply app/grails-app/services/de/laser/SubscriptionService.groovy:1712]: Why not? Every TitleInstance have id

Comment (croqueGrec09) [reply app/grails-app/services/de/laser/SubscriptionService.groovy:1712]: Yes, but you check if an id is in an id. That should raise a TypeMismatchException.

Comment (anjaalbinnrw) [reply app/grails-app/services/de/laser/SubscriptionService.groovy:1712]: Das sollte doch eigentlich beim Testen aufgefallen sein!?

File: app/grails-app/services/de/laser/SubscriptionService.groovy [hunk 3]
```diff
@@ -1707,9 +1707,9 @@ class SubscriptionService {
                     ((colMap.printIdentifierCol >= 0 && cols[colMap.printIdentifierCol].trim().isEmpty()) || colMap.printIdentifierCol < 0)) {
             } else {
 
-                TitleInstance tiObj = TitleInstance.executeQuery('select ti from TitleInstance ti join ti.ids ident where ident.ns in :namespaces and ident.value = :value', [namespaces:idCandidate.namespaces, value:idCandidate.value])
-                if (tiObj) {
-                    IssueEntitlement issueEntitlement = issueEntitlements.find { it.tipp.title.id == tiObj.id }
+                List<TitleInstance> titleIds = TitleInstance.executeQuery('select ti.id from TitleInstance ti join ti.ids ident where ident.ns in :namespaces and ident.value = :value', [namespaces:idCandidate.namespaces, value:idCandidate.value])
```
Comment (croqueGrec09) [inline app/grails-app/services/de/laser/SubscriptionService.groovy:1710]: That gives a ClassCastException

**Pull request merged**